lib/ldap_connection.rb

```diff
@@ -49,6 +49,11 @@ def uidnumber_by_username(username)
     raise "No uidnumber for uid #{username}"
   end
 
+  def test_connection(hostname)
+    @ldap_host = hostname
+    connection.bind ? true : false
```

Comment:
never use ternary operator with true :  false ....
what does #bind return? it might be ok to just call #bind and also use it as return value. maybe you have to use .present?